### PR TITLE
Basic implementation of column reordering

### PR DESCRIPTION
**Pull request by kumarankit1234**
i. Reordering of columns

### CODE REVIEW

**Comment by samrith-s** [comment]
@kumarankit1234 Let's implement this using https://github.com/atlassian/react-beautiful-dnd

This will help us with a lot of stuff, and also in column-resizing.

We either use that, or if we do not want to use additional packages, we should clone the element rather than using native ghost preview.

File: src/components/Rows/Cells/HeaderCell.js [hunk 1]
```diff
@@ -49,6 +58,10 @@ export default class HeaderCell extends PureComponent {
         })}
         style={getCellStyle(style, isSticky)}
         onClick={this.handleSort}
+        draggable={true}
```
Comment (dmitriy-kudelko) [inline src/components/Rows/Cells/HeaderCell.js]: Very minor: passing true is redundant.

**Review by dmitriy-kudelko — CHANGES_REQUESTED** [review]
@kumarankit1234 , I think it'd be cool to do a couple of more things:
- to add a mechanism to prevent re-ordering of certain columns (such as checkbox column for instance). This can be done by providing a certain prop: e.g `reordarable={false}`
- to disable vertical movement while dragging (only allow x axis)
- to drag the whole column rather than just the header
- to add some custom icon/cursor indicating that a column is draggable. Not sure what's the best way to do this, but currently it's difficult to guess that a column can be re-ordered.

**Comment by kumarankit1234** [comment]
@samrith-s 
> We either use that, or if we do not want to use additional packages, we should clone the element rather than using native ghost preview.

Currently we are not showing any preview of the column being reordered. That I will do later once the basic integration of reordering is done with the existing Table.

**Comment by kumarankit1234** [comment]
@dmitriy-kudelko 
> to add a mechanism to prevent re-ordering of certain columns (such as checkbox column for instance). This can be done by providing a certain prop: e.g reordable={false}

This will prevent a column from being reordered. But what will happen in the case when reordering of other columns will have to reorder a column whose reordable={false} is set.?

**Comment by dmitriy-kudelko** [comment]
@kumarankit1234 ,

> This will prevent a column from being reordered. But what will happen in the case when reordering of other columns will have to reorder a column whose reordable={false} is set.?

From the UI perspective I think what should happen is that when you drag any other column over a not re-ordarable one, the cursor should change to disabled and drop event shouldn't fire.

**Comment by kumarankit1234** [comment]
@dmitriy-kudelko As discussed with @samrith-s For now if we are not allowing reOrderable props. For now we will prevent the reodering of the first column if it is a checkbox column. Later we will think of a better way to handle reordering of columns when one of them has reOrderable false.

**Comment by kumarankit1234** [comment]
Also 
> to drag the whole column rather than just the header

Not an important requirement for now. Will do this later.

**Review by samrith-s — APPROVED** [review]
Basic implementation looks good.

**Pull request merged**